src/tls/base64.h

```diff
@@ -39,6 +39,32 @@ namespace tls
     return decoded;
   }
 
+  inline std::vector<uint8_t> raw_from_b64url(
+    const std::string_view& b64url_string)
+  {
+    std::string b64_string = std::string(b64url_string);
+    for (size_t i = 0; i < b64_string.size(); i++)
+    {
+      switch (b64_string[i])
+      {
+        case '-':
+          b64_string[i] = '+';
+          break;
+        case '_':
+          b64_string[i] = '/';
+          break;
+      }
+    }
+    auto padding =
+      b64_string.size() % 4 == 2 ? 2 : b64_string.size() % 4 == 3 ? 1 : 0;
+    while (padding > 0)
+    {
+      b64_string.push_back('=');
+      padding--;
+    }
```

Comment:
I think this padding calculation is missing the `size % 4 == 1` case? And there must be a neater way to express this, I think:

```suggestion
    // Add =s chars to pad string to a multiple of 4
    const auto padding = 3 - ((b64_string.size() - 1) % 4);
    b64_string += std::string(padding, '=');
```

We have some unit tests for b64 encode/decode in `tls/test/main.cpp` - can you add one for `raw_from_b64url` testing different padding sizes?

Reply:
Alternatively, perhaps the most readable:

```suggestion
    while (b64_string.size() % 4 != 0)
    {
      b64_string.push_back('=');
    }
```

Reply:
https://en.wikipedia.org/wiki/Base64#Output_padding. You pad at most two characters.

Reply:
I added a test case for base64url for different sizes, like for base64.